assets/components/src/plugin-settings/index.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * WordPress dependencies.
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { Component, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import { SectionHeader, Notice } from '../';
+import SettingsSection from './SettingsSection';
+
+class PluginSettings extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			inFlight: false,
+			settings: {},
+			error: null,
+		};
+	}
+
+	fetchSettings = () => {
+		const { pluginSlug } = this.props;
+		this.setState( { inFlight: true } );
+		apiFetch( { path: `/${ pluginSlug }/v1/settings-list` } )
+			.then( settings => {
+				this.setState( { settings, error: null } );
+			} )
+			.catch( error => {
+				this.setState( { error } );
+			} )
+			.finally( () => {
+				this.setState( { inFlight: false } );
+			} );
+	};
+
+	componentDidMount() {
+		this.fetchSettings();
+	}
+
+	getSettingsValues = sectionKey => {
+		return (
+			this.state.settings[ sectionKey ]?.reduce( ( map, setting ) => {
+				map[ setting.key ] = setting.value;
+				return map;
+			}, {} ) || {}
+		);
+	};
+
+	handleSettingChange = sectionKey => ( key, value ) => {
+		const sectionSettings = [ ...this.state.settings[ sectionKey ] ];
```

Comment:
`this.state.settings[ sectionKey ]` is already an array, why should it be coerced into array like so?

Reply:
This is not meant for typecasting, the spread operator is commonly used for cloning. From [react docs](https://reactjs.org/docs/react-component.html#state):

> Never mutate this.state directly, as calling setState() afterwards may replace the mutation you made. Treat this.state as if it were immutable.

Reply:
Ah, got it. One of the caveats of class-based React ;)

Reply:
Ah, got it. One of the caveats of class-based React ;)